coherent_rates/isf.py

```diff
@@ -185,11 +219,180 @@ def get_boltzmann_isf(
+    sd = np.std(isf_data, axis=0, dtype=np.complex128)
+    return {
+        "data": mean,
+        "basis": times,
+        "standard_deviation": sd,
+    }
+
+
+def fit_abs_isf_to_gaussian(
+    values: ValueList[_BT0],
+) -> GaussianFitData:
+    times = values["basis"]
+    xdata1 = BasisUtil(times).nx_points
+
+    ydata = get_measured_data(values["data"], "abs")
+
+    def gauss(
+        x: np.ndarray[Any, np.dtype[np.float64]],
+        a: float,
+        b: float,
+    ) -> np.ndarray[Any, np.dtype[np.float64]]:
+        return a * np.exp(-1 * np.square(x / b) / 2)
+
+    parameters, covariance = curve_fit(gauss, xdata1, ydata)
+    fit_A = parameters[0]
+    fit_B = parameters[1]
+    dt = times.times[1]
+
+    return GaussianFitData(
+        fit_A,
+        np.sqrt(covariance[0][0]),
+        fit_B * dt,
+        np.sqrt(covariance[1][1]) * dt,
+    )
+
+
+def fit_abs_isf_to_double_gaussian(
+    values: ValueList[_BT0],
+) -> DoubleGaussianFitData:
+    times = values["basis"]
+    xdata1 = BasisUtil(times).nx_points
+
+    ydata = get_measured_data(values["data"], "abs")
+
+    def gauss(
+        x: np.ndarray[Any, np.dtype[np.float64]],
+        a1: float,
+        b1: float,
+        a2: float,
+        b2: float,
+    ) -> np.ndarray[Any, np.dtype[np.float64]]:
+        return a1 * np.exp(-1 * np.square(x / b1) / 2) + a2 * np.exp(
+            -1 * np.square(x / b2) / 2,
+        )
+
+    parameters, covariance = curve_fit(gauss, xdata1, ydata)
+    fit_A1 = parameters[0]
+    fit_B1 = parameters[1]
+    fit_A2 = parameters[2]
+    fit_B2 = parameters[3]
+    dt = times.times[1]
+
+    return DoubleGaussianFitData(
+        fit_A1,
+        np.sqrt(covariance[0][0]),
+        fit_B1 * dt,
+        np.sqrt(covariance[1][1]) * dt,
+        fit_A2,
+        np.sqrt(covariance[2][2]),
+        fit_B2 * dt,
+        np.sqrt(covariance[3][3]) * dt,
+    )
+
+
+def truncate_value_list(
+    values: ValueList[EvenlySpacedTimeBasis[int, int, int]],
+    index: int,
+) -> ValueList[EvenlySpacedTimeBasis[int, int, int]]:
+    data = values["data"][0 : index + 1]
+    new_times = EvenlySpacedTimeBasis(index + 1, 1, 0, values["basis"].times[index])
+    return {
+        "basis": new_times,
+        "data": data,
+    }
+
+
+class MomentumBasis(FundamentalBasis[Any]):
+    def __init__(self, k_points: np.ndarray[Any, np.dtype[np.float64]]) -> None:
+        self._k_points = k_points
+        super().__init__(k_points.size)  # type: ignore
+
+    @property
+    def k_points(self) -> np.ndarray[Any, np.dtype[np.float64]]:
+        return self._k_points
+
+
+def get_ak_data_1d(
+    system: PeriodicSystem,
+    config: PeriodicSystemConfig,
+    *,
+    k_points: list[int] | None = None,
+) -> ValueList[MomentumBasis]:
+    mass_ratio = system.mass / atomic_mass
+    times = EvenlySpacedTimeBasis(100, 1, 0, 1.5e-14 * np.power(mass_ratio, 0.6))
+    length = get_potential(system, config)["basis"].delta_x_stacked[0]
```

Comment:
declare closer to where it is used, and I would call it delta_x.
Also should be able to use hamiltonian["basis"][0] ?